### PR TITLE
spiutils: Add InactiveSegmentsInfo and RebootRequest/Response messages.

### PR DESCRIPTION
**Pull request by osenft**

### CODE REVIEW

File: shared-lib/spiutils/src/protocol/firmware.rs [hunk 1]
```diff
@@ -322,3 +437,104 @@ impl ToWire for WriteChunkResponse {
         Ok(())
     }
 }
+
+// ----------------------------------------------------------------------------
+
+wire_enum! {
+    /// When to perform the reboot.
+    pub enum RebootTime: u8 {
```
Comment (dann21) [inline shared-lib/spiutils/src/protocol/firmware.rs:445]: Similarly to Johnathan's earlier comment, would this be better phrased as an Option?

File: shared-lib/spiutils/src/protocol/firmware.rs [hunk 2]
```diff
@@ -322,3 +437,104 @@ impl ToWire for WriteChunkResponse {
...
+        w.write_be(self.time.to_wire_value())?;
+        Ok(())
+    }
+}
+
+// ----------------------------------------------------------------------------
+
+wire_enum! {
+    /// The result of a reboot request.
+    pub enum RebootResult: u8 {
```
Comment (dann21) [inline shared-lib/spiutils/src/protocol/firmware.rs:489]: Maybe an Option?

**Review by dann21 — APPROVED** [review]
LGTM, modulo a couple of comments about maybe using an Option instead of an enum that has an Unknown value.

File: shared-lib/spiutils/src/protocol/firmware.rs (hunk 2 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/firmware.rs:489]: Actually, we don't need or want the `Unknown` value. I removed it. Good catch!

File: shared-lib/spiutils/src/protocol/firmware.rs (hunk 1 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/firmware.rs:445]: Actually, we don't need or want the `Unknown` value. I removed it. Good catch!

**Review by dann21 — APPROVED** [review]

**Pull request merged**